samcli/lib/sync/infra_sync_executor.py

```diff
@@ -221,6 +222,7 @@ def _auto_skip_infra_sync(
         packaged_template_path: str,
         built_template_path: str,
         stack_name: str,
+        parameter_overrides: dict = {},
```

Comment:
Apologies that I didn't see this earlier but python has weird behavior with default values of collection type parameters. (see: https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html)

Either remove the default value here, or set the default to `None` and instantiate inside the function with `parameter_overrides = parameter_overrides or {}`.

And please add this parameter to the function documentation below.

Reply:
Good point, forgot to add it to the docstring. Also was not aware of that behaviour, good to know